engine/src/execution_graph/logic_controllers/CacheData.cpp

```diff
@@ -30,6 +30,20 @@ std::string randomString(std::size_t length) {
 	return random_string;
 }
 
+std::string MetadataDictionary::get_value(std::string key) {
+    if (!this->has_value(key)) {
```

Comment:
For this function, if the key does not exist, please return empty string. No need to log a warning